06_neighborhoods/Transportation/transit_trips_cost_city_subgroups.R

```diff
@@ -0,0 +1,789 @@
+# actually see something; above 100, we have already explored outliers above)
+ggplot(subset(trips_by_race_19, index_transit_trips>0), aes(x = index_transit_trips, fill = subgroup)) +
+  geom_histogram(binwidth = 1) +
+  labs(title = "Histogram of Transit Trips by Race Category, 2019",
+       x = "Transit Trips",
+       y = "Frequency") +
+  facet_wrap(~subgroup, scales = "fixed")
+
+# look at summary stats
+summary_by_race <- trips_by_race_19 %>%
+  group_by(subgroup) %>%
+  summarise(
+    count = n(),
+    mean = mean(index_transit_trips, na.rm = TRUE),
+    median = median(index_transit_trips, na.rm = TRUE),
+    min = min(index_transit_trips, na.rm = TRUE),
+    max = max(index_transit_trips, na.rm = TRUE),
+    sd = sd(index_transit_trips, na.rm = TRUE)
+  )
+
+print(summary_by_race)
+
+# examine outliers
+trips_by_race_2019_outliers <- trips_by_race_19 %>% 
+  filter(index_transit_trips>100) 
+# Nothing unexpected
+
+# Use stopifnot to check if all values are non-negative
+stopifnot(min(trips_by_race_19$index_transit_trips, na.rm = TRUE) >= 0)
+# Good to go
+
+
+
+###################################################################
+
+# 5. Export
+
+
+# Combine the All values with the Subgroup values by appending
+cost_race_15 <- bind_rows(cost_all_15, cost_by_race_15)
+cost_race_19 <- bind_rows(cost_all_19, cost_by_race_19)
+trips_race_15 <- bind_rows(trips_all_15, trips_by_race_15)
+trips_race_19 <- bind_rows(trips_all_19, trips_by_race_19)
+
+
+# add a variable for the year of the data
+cost_race_15 <- cost_race_15 %>%
+  mutate(
+    year = 2015
+  )
+cost_race_19 <- cost_race_19 %>%
+  mutate(
+    year = 2019
+  )
+trips_race_15 <- trips_race_15 %>%
+  mutate(
+    year = 2015
+  )
+trips_race_19 <- trips_race_19 %>%
+  mutate(
+    year = 2019
+  )
+
+# Combine the two years into one overall files for both variables
+transit_cost_subgroup_city <- rbind(cost_race_15, cost_race_19)
+transit_trips_subgroup_city <- rbind(trips_race_15, trips_race_19)
+
+
+# Keep variables of interest and order them appropriately
+transit_trips_subgroup_city <- transit_trips_subgroup_city %>%
+  select(year, state, place, subgroup_type, subgroup, index_transit_trips, index_transit_trips_quality) %>%
+  arrange(year, state, place, subgroup_type, subgroup)
+transit_cost_subgroup_city <- transit_cost_subgroup_city %>%
+  select(year, state, place, subgroup_type, subgroup, index_transportation_cost, index_transportation_cost_quality) %>%
+  arrange(year, state, place, subgroup_type, subgroup)
+
+
+# Keep only our cities of interest (486 of them from the places crosswalk)
+places <- read_csv("geographic-crosswalks/data/place-populations.csv")
+# keep the most recent year
+places <- places %>%
+  filter(year == 2019) %>%
+  select(state, place)
+
+# left join to limit to places of interest
+transit_trips_subgroup_city <- left_join(places, transit_trips_subgroup_city, by = c("state", "place"))
```

Comment:
The final file has 3,867 observations. Since each place should have 4 rows times 2 years the total observations should be 3,888 (assuming 486 valid places). I would suggest running a frequency to see which place has less than 8 to diagnose what is going wrong.

Reply:
Thank you for catching - these were missing data that were not being accounted for in the final output. Updating accordingly so the output files account for every valid place & subgroup group

Reply:
Yes -- two place in Georgia (place 49008 and 72122) were missing in the source data entirely, and place 37692 in California was dropping it's NA obs in the subgroup expansion

Reply:
Nice. Thanks.